charts/camunda-platform/templates/operate/deployment.yaml

```diff
@@ -46,6 +46,10 @@ spec:
               value: "identity-auth"
             - name: SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUERURI
               value: {{ include "camundaPlatform.authIssuerBackendUrl" . | quote }}
+            {{- if .Values.global.identity.auth.jwksUrl }}
```

Comment:
When does this `global.identity.auth.jwksUrl` value get set? The reason for the approach we took in our fix (adding an additional environment variable) was we could not find a definitive answer as to whether it was safe to always use the SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI variable without breaking other configs.

When I look in the helm chart readme, it seems like this value is always going to be present:
https://github.com/camunda/camunda-platform-helm/blob/d5bc31322c60be6fc32a755c2a1c99b7379e5448/charts/camunda-platform/README.md?plain=1#L458, which would suggest that for any identity config (not just MS Entra), this would cause operate to use this environment variable instead of the path from the config concatenated with the issuer URL.

Are we absolutely sure about the conditions under which this identity value will be set, which in turn will trigger this environment variable and override the behavior and that it won't break other customers configurations?

Reply:
It can be present for any identity config - but to be honest I don't see an issue here. If the property is present, doesn't it also mean that the customer specifically instructed us to use that specific JWKS URL?

I also tested with the default Keycloak configuration where this value is set to `""`: the condition is not triggered and Operate uses the default URL ending with `/protocol/openid-connect/certs`.

Reply:
> If the property is present, doesn't it also mean that the customer specifically instructed us to use that specific JWKS URL?

This was the question that came up originally during the issue that we don't have a clear answer to. When the identity profile is configured, the code was specifically not looking at SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI (which I presume comes from `global.identity.auth.jwksUrl`)

The intent of the CAMUNDA_OPERATE_USE_SPRING_SECURITY_JWK_URI was a short-term fix that a user would explicitly add if they were using Microsoft Entra or some other OIDC back-end that had a format not supported by the existing behavior of default path concatenated with the `issuer` URL. We did not want to break any existing configs by switching over the behavior to always just use SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI unless we could get clarification that was the correct behavior in all cases.

I guess the question is, is `global.identity.auth.jwksUrl` always set when the identity profile is active? If so, and you want us to look at that value whenever present, it doesn't make sense for us to have the CAMUNDA_OPERATE_USE_SPRING_SECURITY_JWK_URI at all, and we should update the code accordingly. But right now i think there are still open questions about when exactly we should use the value in SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI vs using the default path concatenated with the `issuer` URL.

Reply:
The code has been refactored to use `SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI` and removed this intermediate environment variable, so there shouldn't need to be any changes to the helm chart now

https://github.com/camunda/operate/pull/6362